src/curve_fitting.py

```diff
@@ -0,0 +1,272 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import re
+from gettext import gettext as _
+
+from gi.repository import Adw, GObject, Graphs, Gtk
+
+from graphs import ui
+from graphs.canvas import Canvas
+from graphs.data import Data
+from graphs.item import Item
+from graphs.utilities import preprocess, string_to_function
+
+import numpy
+
+from scipy.optimize import curve_fit
+
+
+@Gtk.Template(resource_path="/se/sjoerd/Graphs/ui/curve_fitting.ui")
+class CurveFittingWindow(Adw.Window):
+    __gtype_name__ = "CurveFittingWindow"
+    equation_entry = Gtk.Template.Child()
+    fitting_params = Gtk.Template.Child()
+    toast_overlay = Gtk.Template.Child()
+    text_view = Gtk.Template.Child()
+
+    def __init__(self, application, item):
+        super().__init__(
+            application=application, transient_for=application.get_window(),
+
+        )
+        ui.bind_values_to_settings(
+            self.get_application().get_settings("curve-fitting"), self)
+        canvas = Canvas(application)
+        self.param = []
+
+        # Set data item that contain the curves
+        self.curves = Data(application)
```

Comment:
would a simple list suffice here? Using the whole Data class seems a bit overkill

Reply:
I'm mainly using the Data class to bind these to the canvas that is used:
`        self.curves.bind_property("items", canvas, "items", 2)`

If you have a suggestion to bind the canvas to the used items without doing this, then that would probably be the preferred solution, since the other properties of this class are not really used.